### PR TITLE
feat(route): Add Chub

### PR DESCRIPTION
**Pull request by flameleaf**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/chub/characters
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/routes/chub/characters.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    categories: ['new-media'],
+    example: '/chub/characters',
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
```
Comment (TonyRL) [inline lib/routes/chub/characters.ts:16]:
```suggestion
```

File: lib/routes/chub/characters.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    name: 'Characters',
+    maintainers: ['flameleaf'],
+    handler,
+};
+
+
+async function handler() {
+    const host = `https://www.chub.ai/characters`;
```
Comment (TonyRL) [inline lib/routes/chub/characters.ts:24]:
```suggestion
    const host = 'https://www.chub.ai/characters';
```

File: lib/routes/chub/characters.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        method: 'get',
+        url,
+        headers,
+    });
+    const data = response.data;
+    const nodes = data.nodes;
+
+    return {
+        allowEmpty: true,
+        title: `Chub`,
```
Comment (TonyRL) [inline lib/routes/chub/characters.ts:37]:
```suggestion
        title: 'Chub',
```

File: lib/routes/chub/characters.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+async function handler() {
+    const host = `https://www.chub.ai/characters`;
+    const url = `https://api.chub.ai/api/characters/search?query=&first=200&page=1&sort=last_activity_at&asc=false&include_forks=false&nsfw=true&nsfl=true&nsfw_only=false&require_images=false&require_example_dialogues=false&require_alternate_greetings=false&require_custom_prompt=false&exclude_mine=false&venus=true&chub=true&min_tokens=50&require_expressions=false&require_lore=false&mine_first=false&require_lore_embedded=false&require_lore_linked=false&inclusive_or=false&recommended_verified=false`;
+    const headers = { Accept: 'application/json' };
+    const response = await got({
+        method: 'get',
+        url,
+        headers,
+    });
```
Comment (TonyRL) [inline lib/routes/chub/characters.ts:31]: Please use [ofetch](https://docs.rsshub.app/joinus/new-rss/start-code#basic-code) with [query](https://github.com/unjs/ofetch?tab=readme-ov-file#%EF%B8%8F-adding-query-search-params) option.

File: lib/routes/chub/characters.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+    return {
+        allowEmpty: true,
+        title: `Chub`,
+        link: host,
+        item: nodes
+            .map((item) => ({
+                    title: item.name,
+                    description: `${item.tagline}<br><br>${item.description}<br><br>${item.topics}`,
+                    pubDate: parseDate(item.lastActivityAt),
```
Comment (TonyRL) [inline lib/routes/chub/characters.ts:43]:
```suggestion
                    pubDate: parseDate(item.createdAt),
                    updated: parseDate(item.lastActivityAt),
```

File: lib/routes/chub/characters.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    const nodes = data.nodes;
+
+    return {
+        allowEmpty: true,
+        title: `Chub`,
+        link: host,
+        item: nodes
+            .map((item) => ({
+                    title: item.name,
+                    description: `${item.tagline}<br><br>${item.description}<br><br>${item.topics}`,
```
Comment (TonyRL) [inline lib/routes/chub/characters.ts:42]: Please put `topics` to `category` instead.

File: lib/routes/chub/characters.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+};
+
+
+async function handler() {
+    const hostURL = 'https://www.chub.ai/characters';
+    const apiURL = 'https://api.chub.ai/api/characters/search?query=';
+    const data = await ofetch(apiURL, {
+        headers: { Accept: 'application/json' },
+        query: {
+            first: 200,
```
Comment (TonyRL) [inline lib/routes/chub/characters.ts:21]:
```suggestion
    const apiURL = 'https://api.chub.ai/api/characters/search';
    const data = await ofetch(apiURL, {
        headers: { Accept: 'application/json' },
        query: {
            query: '',
            first: 200,
```

**Review by Sidon94 — APPROVED** [review]

**Pull request merged**